lxd/auth/authorization_objects.go

```diff
@@ -0,0 +1,183 @@
+package auth
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+func ObjectUser(userName string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeUser, userName)
+}
+
+func ObjectServer() string {
+	return fmt.Sprintf("%s:lxd", ObjectTypeServer)
+}
+
+func ObjectCertificate(fingerprint string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeCertificate, fingerprint)
+}
+
+func ObjectStoragePool(storagePoolName string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeStoragePool, storagePoolName)
+}
+
+func ObjectProject(projectName string) string {
+	return fmt.Sprintf("%s:%s", ObjectTypeProject, projectName)
+}
+
+func ObjectImage(projectName string, imageFingerprint string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeImage, projectName, imageFingerprint)
+}
+
+func ObjectImageAlias(projectName string, aliasName string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeImageAlias, projectName, aliasName)
+}
+
+func ObjectInstance(projectName string, instanceName string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeInstance, projectName, instanceName)
+}
+
+func ObjectNetwork(projectName string, networkName string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeNetwork, projectName, networkName)
+}
+
+func ObjectNetworkACL(projectName string, networkACLName string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeNetworkACL, projectName, networkACLName)
+}
+
+func ObjectNetworkZone(projectName string, networkZoneName string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeNetworkZone, projectName, networkZoneName)
+}
+
+func ObjectProfile(projectName string, profileName string) string {
+	return fmt.Sprintf("%s:%s_%s", ObjectTypeProfile, projectName, profileName)
+}
+
+func ObjectStorageBucket(projectName string, poolName string, bucketName string) string {
+	return fmt.Sprintf("%s:%s_%s_%s", ObjectTypeStorageBucket, projectName, poolName, bucketName)
+}
+
+func ObjectStoragePoolVolume(projectName string, poolName string, volumeType string, volumeName string) string {
+	return fmt.Sprintf("%s:%s_%s_%s_%s", ObjectTypeStoragePoolVolume, projectName, poolName, volumeType, volumeName)
+}
```

Comment:
I can't remember if there was a previous discussion about these object names. They work fine as they are but if we want to be able to reliably deconstruct these into their component parts then we may not always be able to use an underscore as a delimiter. 

Are there any characters or character sequences that are not allowed in any LXD API resource? A forward slash perhaps? Another alternative would be to use the API URL directly.

Reply:
@monstermunchkin I've just realised that this is why you used `/` as a delimiter for storage volume and buckets (after the project name, which cannot contain an underscore). I'll update this to be the same.

Reply:
I like the idea of using parameter encoding. `storage_pool_volume:project=default&storage_pool=default&storage_volume_type=custom&storage_volume_name=vol1` is a lot more explicit (albeit a lot more verbose) than having `storage_pool_volume:default_default/custom/vol1`.

Reply:
Ala https://github.com/canonical/lxd/blob/main/shared/api/url.go#L34-L45

Reply:
I think this is a good middle ground so I'll go for that. Thanks.

Reply:
Ok that's done now.